dlt/pipeline/pipeline.py

```diff
@@ -1026,6 +1028,8 @@ def _resolve_loader_file_format(
             possible_file_formats = [f for f in dest_caps.supported_staging_file_formats if f in stage_caps.supported_loader_file_formats]
         if not file_format:
             if not stage_caps:
+                if not dest_caps.preferred_loader_file_format:
```

Comment:
maybe we should really add staging automatically if destination is used that requires it? it could happen in `_set_destination` and we can issue warning that we did that. WDT? this simplifies the code and we can also fix dlt init (I can do that)